sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/AddTryCatchToMethodCallRecipe.java

```diff
@@ -0,0 +1,328 @@
+        this.methodPattern = methodPattern;
+        this.excludeOwner = excludeOwner;
+        this.catchTemplateString = catchTemplateString;
+        this.fullyQualifiedExceptionName = fullyQualifiedExceptionName;
+    }
+
+
+    @Override
+    public @NlsRewrite.DisplayName @NotNull String getDisplayName() {
+        return "Add try-catch to method";
+    }
+
+    @Override
+    public @NlsRewrite.Description @NotNull String getDescription() {
+        return "Surrounds calls to the target method in a custom try-catch block.";
+    }
+
+    /**
+     * Method to return the visitor that performs the checks and changes
+     *
+     * @return Returns the visitor that performs the checks and changes
+     */
+    @Override
+    public @NotNull TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new AddTryCatchVisitor();
+    }
+
+    /**
+     * Visitor that performs the checks and changes
+     */
+    private class AddTryCatchVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        private final MethodMatcher methodMatcher = new MethodMatcher(methodPattern, true);
+
+        /**
+         * Overridden visitBlock method performs the changes to methods filtered by visitMethodCall.
+         */
+        @Override
+        public J.@NotNull Block visitBlock(J.@NotNull Block block, @NotNull ExecutionContext context) {
+            J.Block body = super.visitBlock(block, context);
+
+            // Get the method that needs to be changed
+            MethodCall method = getCursor().pollMessage("METHOD");
+            if (method == null) {
+                return body;
+            }
+
+            //Get the parents of the method
+            Tree parent = getCursor().pollMessage("PARENT");
+            // Get the first statement parent of method
+            Statement parentStatement = getCursor().pollMessage("STATEMENT");
+
+            JavaTemplate tryCatchTemplate = JavaTemplate.builder("try { int a = null; a = 3; } " + catchTemplateString)
+                    .imports(fullyQualifiedExceptionName)
+                    .build();
+
+            // Create an empty block to apply the try-catch template based off the cursor values from the main body
+            // This should create the correct formatting.
+            J.Block b = J.Block.createEmptyBlock();
+            b = tryCatchTemplate.apply(new Cursor(getCursor(),b), b.getCoordinates().firstStatement());
+            int parentIndex;
+
+            // Extract the try-catch block and dummy elements
+            J.Try _try = (J.Try) b.getStatements().get(0);
+            J.VariableDeclarations dummyVarDec = (J.VariableDeclarations) _try.getBody().getStatements().get(0);
+            J.Assignment dummyAssignment = (J.Assignment) _try.getBody().getStatements().get(1);
+
+            if (_try.getCatches().isEmpty()) {
+                // The catch template was incorrect, recipe is unsafe.
```

Comment:
Do we want reporting when a recipe is unsafe or doesn't work as expected? This could show a bug in our or user code

Reply:
@alzimmermsft I personally believe that ultimately our testing will be the key factor in determining whether or not we can trust a recipe. I don't think we need external reporting at this point.